src/hooks/useCreatePlayer.ts

```diff
@@ -0,0 +1,30 @@
+import { useAppDispatch, useAppSelector } from "../state/reduxHooks";
+import axios from "axios";
+import { useEffect } from "react";
+import { API_URL } from "../settings";
+import { setPlayerName } from "../state/playerSlice";
+import { joinGameRoomAsHostAction } from "../state/socketActions/joinGameRoomAction";
+
+export function useCreatePlayer() {
+    const { name } = useAppSelector(state => state.player);
+    const { code } = useAppSelector(state => state.game);
+    const dispatch = useAppDispatch();
+    useEffect(() => {
+        if (!name || !code) return;
+        axios
+            .post(`${API_URL}/players`,
+                {
+                    "name": name,
+                    "joinCode": code
+                })
+            .then(response => {
+                // Not sure what we should do with the response
+                console.log(response.status);
+            })
+            .catch(reason => {
+                // TODO: Handle errors better
```

Comment:
This is probably where we will handle the case of the name being already taken. We should test what response the API gives on that scenario (what status code it is), and then send the alert from here

Reply:
Makes sense, I added an alert here